vector_setters.go

```diff
@@ -151,21 +363,63 @@ func (vec *vector) setList(rowIdx C.idx_t, val any) {
 	childVector := &vec.childVectors[0]
 	for i, entry := range list {
 		offset := C.idx_t(i) + childVectorSize
-		childVector.setFn(childVector, offset, entry)
+		err := childVector.setFn(childVector, offset, entry)
+		if err != nil {
+			return err
+		}
 	}
+	return nil
 }
 
-func (vec *vector) setStruct(rowIdx C.idx_t, val any) {
-	m := val.(map[string]any)
+func setStruct[S any](vec *vector, rowIdx C.idx_t, val S) error {
+	var m map[string]any
+	switch v := any(val).(type) {
+	case map[string]any:
+		m = v
+	default:
+		//TODO type.Kind() == reflect.Map??
```

Comment:
I don't think we should mix `STRUCT` and `MAP` here. Or maybe I misunderstand your `TODO`?

Reply:
Oh I just wanted to support more kinds of maps besides `map[string]any`, which I think will be nice. Since duckdb structs are not always known at compiletime, I think that allowing the user to use a types map would give the user more flexibility when working with duckdb structs. Only allowing go structs is very restrictive (since it has to be known at compile time), and users might want more type information than having to use a `map[string]any`

Reply:
> allowing the user to use a types map would give the user more flexibility when working with duckdb structs. Only allowing go structs is very restrictive

Ah, got you! 👍 I think for this PR we can turn it into a `FIXME` and potentially add this later. It does sound like a nice idea to give flexibility.

Reply:
should I implement this? or do you disagree

Reply:
Ah sorry your previous message hadn't shown up on my screen yet, I thought github always live-updated things but that never really seems to work. Made it a fixme